src/common/transformations/src/transformations/op_conversions/group_query_attention_decomposition.cpp

```diff
@@ -0,0 +1,258 @@
+                                     ov::element::i64);
+    auto hori_range = register_new_node<v0::Unsqueeze>(mask_per_line_node, zero);
+    auto vert_range = register_new_node<v0::Unsqueeze>(mask_per_line_node, one);
+    auto triu = register_new_node<v1::Greater>(hori_range, vert_range);
+    auto typed_zero = register_new_node(v0::Constant::create(T, ov::Shape{}, {0}));
+    // cf. make_attention_mask@src\plugins\intel_gpu\tests\common\subgraphs_builders.hpp
+    std::shared_ptr<ov::Node> minus_inf = nullptr;
+    if (T == ov::element::f32)
+        minus_inf = register_new_node(v0::Constant::create(T, ov::Shape{}, {-std::numeric_limits<float>::infinity()}));
+    else if (T == ov::element::f16)
+        minus_inf =
+            register_new_node(v0::Constant::create(T, ov::Shape{}, {std::numeric_limits<ov::float16>::lowest()}));
+    auto atten_mask = register_new_node<v1::Select>(triu, minus_inf, typed_zero);
+    auto atten_mask_sliced = register_new_node<v8::Slice>(atten_mask, past_sequence_length, seqlens_1d, one, zero);
+
+    std::shared_ptr<ov::Node> qga_output;
+    if (scale != 0.0f) {
+        auto scale_node = register_new_node(v0::Constant::create(T, Shape{}, {scale}));
+        qga_output = register_new_node<v13::ScaledDotProductAttention>(Q, K, V, atten_mask_sliced, scale_node, false);
+    } else {
+        qga_output = register_new_node<v13::ScaledDotProductAttention>(Q, K, V, atten_mask_sliced, false);
+    }
+
+    // transpose the result from (batch_size, num_heads, sequence_length, head_size)
+    // to (batch_size, sequence_length, num_heads * head_size)
+    auto perm = register_new_node(v0::Constant::create(ov::element::i64, ov::Shape{4}, {0, 2, 1, 3}));
+    auto qga_output_transposed = register_new_node<v1::Transpose>(qga_output, perm);
+    auto dim_merge_shape = register_new_node(v0::Constant::create(ov::element::i32, ov::Shape{3}, {0, 0, -1}));
+    auto output = register_new_node<v1::Reshape>(qga_output_transposed, dim_merge_shape, true)->output(0);
+
+    return {output, present_k, present_v};
+}
+
+// make split functions is a copy-past from ONNX FE. TODO: move it to one place
+ov::OutputVector ov::pass::GroupQueryAttentionDecomposition::make_split(const ov::Output<ov::Node>& value,
+                                                                        int64_t num_splits,
+                                                                        int64_t axis) {
+    using namespace ov::op;
+    const auto axis_node = register_new_node(v0::Constant::create(ov::element::i64, ov::Shape{}, {axis}));
+    const auto split = register_new_node<v1::Split>(value, axis_node, num_splits);
+
+    return split->outputs();
+}
+
+std::shared_ptr<ov::Node> ov::pass::GroupQueryAttentionDecomposition::get_dimensions(
+    const std::shared_ptr<ov::op::v3::ShapeOf>& shape,
+    const std::vector<int>& dims) {
+    using namespace ov::op;
+    const auto zero = v0::Constant::create(ov::element::i32, ov::Shape{}, {0});
```

Comment:
minor:
We use i64 in some places and i32 in others. There isn't a restriction to do so but  it's better to align element types, e.g. use i64